### DISCUSSION TITLE
magick command doesn't work in Windows 10

### DISCUSSION THREAD

**Discussion by kalankaboom** [ImageMagick/ImageMagick#4903] [Help]
After first installing the ImageMagick tool, I found that the magick command simply wasn't working.
However I managed to solve the issue by looking at my **System Environment Variables**. ImageMagick automatically adds the directory to the system path, but it adds it all the way at the top. I found that the magick command started working when I deleted the automatically added directory and readded it at the end.
To achieve this, search **"Environment variables"** in the Windows search bar, then click to edit. Then, under the **Advanced** tab, click on **Environment variables**. Under **System Variables**, find **Path**. Click on **Edit**, then delete the automatic directory, and finally add it again by clicking on **New** and pasting in the ImageMagick directory.

**Comment by urban-warrior** [comment]
Adding the path to the front of the system path works for nearly everyone as expected-- excepting perhaps you.  We're clueless why it fails for you.   To confirm, we tried installing the very latest release of ImageMagick on a Windows 11 system and `magick` ran without any issues in the Command Prompt window.

**Reply by audas** [reply]
Unable to get composite to work on windows 10.

I have added the full path absolutely everywhere I can think of and it just wont work.

The STRANGEST part is if I go into command prompt and type "%composite%" then it works. 

Absolutely pulling my hair out by the roots.

**Reply by fmw42** [reply]
Imagemagick is a command line tool.  You need to type your command in the CMD/RUN window.  What exactly did you do?  What is your Imagemagick version?  What do you get from 

```
magick -version

or 

convert -version
```

**Reply by audas** [reply]
Thanks Fred, I was able to use convert etc however I was unable to get compopsite to work. I had the pathway to imagemagick including the convert.exe on the environment path.  Removing that allowed composite to work. I did NOT know that using legacy convert blocked me from using newer features though, I thought it allowed you to continue using new features with the old language.  %FX does not work.

**Reply by fmw42** [reply]
When you install Imagemagick, you get IM 7 by default and have to check the legacy install option in your Windows installer to get IM 6.  If you use convert and composite, you should be getting the IM 6 if you have the legacy components installed.  To use IM 7, you would use magick in place of convert, but magick composite in place of just composite. Similarly for all the other tools (magick identify, magick mogrify, magick montage, etc) all but convert. For convert you just replace it with magick.  Note that only magick has the new features. Magick composite and all the other tools are just the IM 6 tools called from IM 7.   Be sure you properly check the legacy box when doing the install if you want to use IM 6 convert, composite, etc.  And to get new features, you need to be using IM 7 magick. So, I suggest you use magick in place of magick composite. Magick can do anything you do with magick composite and do more and get all the new features.

**Comment by snibgo** [comment]
I thought you solved this, on your other thread https://github.com/ImageMagick/ImageMagick/discussions/5738

> I have been able to add the variable to the environment path and then test it using "%composite%" however it does NOT work without this.

> The STRANGEST part is if I go into command prompt and type "%composite%" then it works.

Well, what is the value of that variable? What does the command ...
```
echo %composite%
```
... say?

The `composite` command is IM v6, which is now very old. I suggest using `magick composite` instead. Even better, use `magick ... -composite ...`.

**Reply by audas** [reply]
Thanks I did solve it. 

I have a HUGE software library built, and something like this is a critical change.

**Comment by rubyFeedback** [comment]
I tried to find more information about windows + imagemagick + bin/magick and the six:- commandline flag.

On linux via KDE konsole and bash, I can display images fine that way. I usually do that via ruby, by first printing some ' ' spaces, to align the image more into the middle.

I now also tested this on windows and I was surprised that cmd.exe shows some strange characters. I assume that cmd.exe may not be useful to display images via bin/magick and six:-

I will continue to do more experiments here eventually, to find out whether six:- works on windows or not. If the official documentation does not mention it, I think it may be useful to briefly explain why this works on some terminals and not on others. Normally when I run ruby, most things work fine on linux, as well as on cmd.exe for windows.

**Comment by snibgo** [comment]
@rubyFeedback: your question is vague, so a response is difficult.

We can list available formats like this:
```
magick -list format |grep -i six
      SIX* rw-   DEC SIXEL Graphics Format
    SIXEL* rw-   DEC SIXEL Graphics Format
```
So `six:` is a file format, that IM can read and write. If you write the contents of an image file to stdout (the console), then what the console displays will depend on the console. Under Windows 11, the command ...
```
magick rose: six:-
```
... reads IM's built-in "rose" image, and writes it to stdout in the `six:` format. The console then displays the image.

For some reason, the `six:` format is not documented at https://imagemagick.org/formats/.
